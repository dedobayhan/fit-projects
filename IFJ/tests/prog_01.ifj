var int vysl;
void main();

int faktorial(int i)
{
	if (i > 1) goto calc;
	if (i < 0) goto error;
	return 1;
calc:
	return i*faktorial(i-1);
error:
	return i; //vraci zaporne cislo
}

void main()
var int param;
{
	param := 5;
	vysl := faktorial(param);
	printstring("Vysledkem funkce faktorial("
						+ int2str(param)
						+ ") je "
						+ int2str(vysl) );
}
