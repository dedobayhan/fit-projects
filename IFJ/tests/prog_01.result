-->./tests/prog_01.ifj<--
TOK_VAR          - var
TOK_INT          - int
TOK_IDENTIFIER   - vysl
TOK_SEMICOLON    - ;
TOK_VOID         - void
TOK_IDENTIFIER   - main
TOK_LBRACKET     - (
TOK_RBRACKET     - )
TOK_SEMICOLON    - ;
TOK_INT          - int
TOK_IDENTIFIER   - faktorial
TOK_LBRACKET     - (
TOK_INT          - int
TOK_IDENTIFIER   - i
TOK_RBRACKET     - )
TOK_LVINCULUM    - {
TOK_IF           - if
TOK_LBRACKET     - (
TOK_IDENTIFIER   - i
TOK_GREATER      - >
TOK_CONSTINT     - 1
TOK_RBRACKET     - )
TOK_GOTO         - goto
TOK_IDENTIFIER   - calc
TOK_SEMICOLON    - ;
TOK_IF           - if
TOK_LBRACKET     - (
TOK_IDENTIFIER   - i
TOK_LESS         - <
TOK_CONSTINT     - 0
TOK_RBRACKET     - )
TOK_GOTO         - goto
TOK_IDENTIFIER   - error
TOK_SEMICOLON    - ;
TOK_RETURN       - return
TOK_CONSTINT     - 1
TOK_SEMICOLON    - ;
TOK_IDENTIFIER   - calc
TOK_LABEL        - :
TOK_RETURN       - return
TOK_IDENTIFIER   - i
TOK_MULTIPLE     - *
TOK_IDENTIFIER   - faktorial
TOK_LBRACKET     - (
TOK_IDENTIFIER   - i
TOK_MINUS        - -
TOK_CONSTINT     - 1
TOK_RBRACKET     - )
TOK_SEMICOLON    - ;
TOK_IDENTIFIER   - error
TOK_LABEL        - :
TOK_RETURN       - return
TOK_IDENTIFIER   - i
TOK_SEMICOLON    - ;
TOK_RVINCULUM    - }
TOK_VOID         - void
TOK_IDENTIFIER   - main
TOK_LBRACKET     - (
TOK_RBRACKET     - )
TOK_VAR          - var
TOK_INT          - int
TOK_IDENTIFIER   - param
TOK_SEMICOLON    - ;
TOK_LVINCULUM    - {
TOK_IDENTIFIER   - param
TOK_ASSIGNMENT   - :=
TOK_CONSTINT     - 5
TOK_SEMICOLON    - ;
TOK_IDENTIFIER   - vysl
TOK_ASSIGNMENT   - :=
TOK_IDENTIFIER   - faktorial
TOK_LBRACKET     - (
TOK_IDENTIFIER   - param
TOK_RBRACKET     - )
TOK_SEMICOLON    - ;
TOK_IDENTIFIER   - printstring
TOK_LBRACKET     - (
TOK_CONSTSTRING  - Vysledkem funkce faktorial(
TOK_PLUS         - +
TOK_IDENTIFIER   - int2str
TOK_LBRACKET     - (
TOK_IDENTIFIER   - param
TOK_RBRACKET     - )
TOK_PLUS         - +
TOK_CONSTSTRING  - ) je 
TOK_PLUS         - +
TOK_IDENTIFIER   - int2str
TOK_LBRACKET     - (
TOK_IDENTIFIER   - vysl
TOK_RBRACKET     - )
TOK_RBRACKET     - )
TOK_SEMICOLON    - ;
TOK_RVINCULUM    - }
