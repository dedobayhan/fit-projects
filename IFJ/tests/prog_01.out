Vysledkem funkce faktorial(5) je 120